test/integration/tests/fuse.ts

```diff
@@ -0,0 +1,112 @@
+    doLogging && console.log(`Environment loaded.`);
+  });
+
+  describe('D3 Fuse Plugin', function () {
+    it('deposit, claim, then withdraw', async function () {
+      const comptroller = await ethers.getContractAt('Unitroller', contractAddresses.rariPool8Comptroller);
+
+      const fD3 = await ethers.getContractAt(
+        'CErc20Delegator',
+        await comptroller.cTokensByUnderlying(contractAddresses.curveD3pool)
+      );
+
+      const d3Whale = '0x16c2bee6f55dab7f494dba643ff52ef2d47fba36';
+
+      await forceEth(d3Whale);
+
+      const d3 = await ethers.getContractAt('IERC20', contractAddresses.curveD3pool);
+
+      const signer = await getImpersonatedSigner(deployAddress);
+
+      doLogging && console.log('Seeding d3');
+      await d3
+        .connect(await getImpersonatedSigner(d3Whale))
+        .transfer(deployAddress, ethers.constants.WeiPerEther.mul(1_000_000));
+
+      doLogging && console.log('Approving d3');
+      await d3.connect(signer).approve(fD3.address, ethers.constants.MaxUint256);
+
+      const d3RewardsBalanceBefore = await contracts.convexD3poolRewards.balanceOf(plugin);
+      const fd3BalanceBefore = await fD3.balanceOfUnderlying(deployAddress);
+
+      doLogging && console.log('Minting fD3');
+      await fD3.connect(signer).mint(ethers.constants.WeiPerEther.mul(1_000_000));
+
+      expect(await contracts.convexD3poolRewards.balanceOf(plugin)).to.be.equal(
+        d3RewardsBalanceBefore.add(ethers.constants.WeiPerEther.mul(1_000_000))
+      );
+      expect(await fD3.balanceOfUnderlying(deployAddress)).to.be.equal(
+        fd3BalanceBefore.add(ethers.constants.WeiPerEther.mul(1_000_000))
+      );
+
+      doLogging && console.log('Redeeming Underlying');
+      await fD3.connect(signer).redeemUnderlying(ethers.constants.WeiPerEther.mul(1_000_000));
+
+      expect(await contracts.convexD3poolRewards.balanceOf(plugin)).to.be.equal(d3RewardsBalanceBefore);
+      expect(await fD3.balanceOfUnderlying(deployAddress)).to.be.equal(fd3BalanceBefore);
+
+      doLogging && console.log('Notifying rewards');
+
+      const operator = await getImpersonatedSigner(await contracts.convexD3poolRewards.operator());
+      await forceEth(await operator.getAddress());
+      await time.increaseTo(await contracts.convexD3poolRewards.periodFinish());
+
+      await contracts.convexD3poolRewards
+        .connect(operator)
+        .queueNewRewards(ethers.constants.WeiPerEther.mul(10_000_000));
+
+      doLogging && console.log('Claiming rewards');
+
+      // Note this is not the right interface but it has the desired function
+      const rewards = await ethers.getContractAt('ConvexPCVDeposit', plugin);
+      await rewards.claimRewards();
+
+      expect(await contracts.crv.balanceOf(contractAddresses.d3poolConvexPCVDeposit)).to.be.at.least(1);
+      expect(await contracts.cvx.balanceOf(contractAddresses.d3poolConvexPCVDeposit)).to.be.at.least(1);
```

Comment:
would like a better check here that compares before and after values

Reply:
yeah because right now the balances are not 0 for instance

Reply:
Yup great callout